cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,325 @@
+package cloudtrail
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	config "github.com/openshift/osdctl/pkg/envConfig"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// LookupEventsoptions struct for holding options for event lookup
+type LookupEventsOptions struct {
+	clusterID      string
+	startTime      string
+	printEventUrl  bool
+	printRawEvents bool
+	printAllEvents bool
+}
+
+// RawEventDetails struct for holding raw event details
+type RawEventDetails struct {
+	EventVersion string `json:"eventVersion"`
+	UserIdentity struct {
+		AccountId      string `json:"accountId"`
+		Type           string `json:"type"`
+		SessionContext struct {
+			SessionIssuer struct {
+				Type     string `json:"type"`
+				UserName string `json:"userName"`
+				Arn      string `json:"arn"`
+			} `json:"sessionIssuer"`
+		} `json:"sessionContext"`
+	} `json:"userIdentity"`
+	EventRegion string `json:"awsRegion"`
+	EventId     string `json:"eventID"`
+}
+
+func newCmdWriteEvents() *cobra.Command {
+	ops := &LookupEventsOptions{}
+	listEventsCmd := &cobra.Command{
+		Use:   "write-events",
+		Short: "Prints out all cloudtrail write events to console",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	listEventsCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	listEventsCmd.Flags().StringVarP(&ops.startTime, "since", "", "", "Start time of the lookup i.e (\"2h\" for starttime 2 hours ago")
+	listEventsCmd.Flags().BoolVarP(&ops.printEventUrl, "url", "u", false, "Generates Url link to cloud console cloudtrail event")
+	listEventsCmd.Flags().BoolVarP(&ops.printRawEvents, "raw-event", "r", false, "Prints the cloudtrail events to the console in raw json format")
+	listEventsCmd.Flags().BoolVarP(&ops.printAllEvents, "all", "A", false, "Prints all cloudtrail write events without filtering")
+	listEventsCmd.MarkFlagRequired("cluster-id")
+	return listEventsCmd
+}
+
+// parseDurationToUTC parses the given startTime string as a duration and subtracts it from the current UTC time.
+// It returns the resulting time and any parsing error encountered.
+func parseDurationToUTC(input string) (time.Time, error) {
+	duration, err := time.ParseDuration(input)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Now().UTC().Add(-duration), nil
+}
+
+// whoami retrieves caller identity information
+func whoami(stsClient sts.Client) (Arn string, AccountId string, err error) {
```

Comment:
Do we need the uppercase? 
```suggestion
func whoami(stsClient sts.Client) (arn string, accountId string, err error) {
```